packages/lib/modules/tokens/TokenInputsValidationProvider.tsx

```diff
@@ -1,17 +1,20 @@
 'use client'
 
 import { useMandatoryContext } from '@repo/lib/shared/utils/contexts'
-import { PropsWithChildren, createContext, useState } from 'react'
+import { PropsWithChildren, createContext, useCallback, useState } from 'react'
 import { Address } from 'viem'
 import { ApiOrCustomToken } from './token.types'
 
 export function useTokenInputsValidationLogic() {
   type ValidationErrorsByToken = Record<Address, string>
   const [validationErrors, setValidationErrors] = useState<ValidationErrorsByToken>({})
 
-  function setValidationError(tokenAddress: Address, value: string) {
-    setValidationErrors({ ...validationErrors, [tokenAddress]: value })
-  }
+  const setValidationError = useCallback((tokenAddress: Address, value: string) => {
+    setValidationErrors(prev => {
+      if (prev[tokenAddress] === value) return prev
+      return { ...prev, [tokenAddress]: value }
+    })
+  }, [])
```

Comment:
`useCallback` to avoid maximum update depth exceeded when calling `setValidationError` from within a `useEffect` as token amounts change

Reply:
this is already in `main` but without the callback, probably because it's not used in `useEffect` yet

but validation is not re-triggered after a refresh (just fill in 0 and refresh the page)

Reply:
@groninge01 fix maybe out of scope for this PR wdyt about new issue?

- https://github.com/balancer/frontend-monorepo/issues/1954